### ISSUE TITLE
Feature Request: Make ossec-logcollector periodically rescan all localfile locations for new files to analyze

### ISSUE THREAD

**Issue by branchnetconsulting** [wazuh/wazuh#195]
## Problem: 

Any time a file matching a localfile location in ossec.conf or agent.conf, is newly created while Wazuh agent is running, the file will not be analyzed by ossec-logcollector until the agent is restarted.  In that new file, none of the log data added to it previous to the agent restart will ever be processed.  For example, given the config below, if I create for the first time log files _/logs/neato.log_ and  _/var/log/interesting.log_, I could pump many megabytes of log content into both of them and the Wazuh agent would not send any of it to the manager until restarted.  Once the agent is restarted, all new content written to those two files will be processed, but anything previous to the restart will remain uninspected.
```
<localfile>
   <log_format>syslog</log_format>
   <location>/logs/*.log</location>
</localfile>

<localfile>
   <log_format>syslog</log_format>
   <location>/var/log/interesting.log</location>
</localfile>
```

## Proposed solution:

Allow ossec-logcollector to be configured to periodically rescan all localfile locations for new files that did not exist at start time, and to immediately start analyzing them.  Optionally allow newly appearing files like this to be analyzed from the start of the file so no log data is missed.

(See [issue 137](https://github.com/wazuh/wazuh/issues/137))

Possible new options in /var/ossec/etc/internal_options.conf to accomplish the above:

```
# Logcollector periodically rescan for new files to analyze (0=disabled, 1=enabled, default=0) 
# Without this, Logcollector will only analyze files that it saw at daemon start time.
logcollector.rescan_for_new_files=1

# Logcollector frequency (in seconds) for rescanning for new files to analyze (default of 300 = rescan every 5 minutes) 
# Only applies when logcollector.rescan_for_new_files=1.  
logcollector.rescan_frequency=300

# Logcollector start analyzing newly appeared files from the beginning (0=disabled, 1=enabled, default=0)
# Only applies when logcollector.rescan_for_new_files=1.  
logcollector.read_new_files_from_start=1
```

Thank you for considering this feature.  It would be especially nice to have when log file names are generated dynamically.

**Comment by vikman90** [comment]
Hi @branchnetconsulting,

Thank you for this proposal. I think that files with wildcards should also be reloaded, just like timestamp-based names.

The parameter `vcheck_files` controls how often those files are reloaded, so we will do the same with paths with wildcards.

We're adding this issue to our roadmap and will keep you updated. I hope to have this new feature ready very soon!

Thank you.

**Comment by branchnetconsulting** [comment]
Hi Vikman,

Yes, that sounds great!  Also please consider reloading non-wildcard files
that had not existed at logcollector start time that may have appeared
since then.  A common place where that has bitten me is analyzing
active-responses.log which is not created until an agent's first active
response event.  I deployed Wazuh agent across a ton of hosts and it was
configured to analyze active-responses.log.  Because that log file didn't
already exist at logcollector start time, the agent just gave up on
watching it until the next restart.  During the periodic file reload, let's
make logcollector also check again on non-wildcard files that previously
were not present, and just like when wildcard files newly appear, start
analyzing those from the beginning of the file instead of starting to tail
at the end of the file.

Thanks!
Kevin

**Comment by vikman90** [comment]
Hi @branchnetconsulting,

This enhancement has been merged into branch 3.6! (https://github.com/wazuh/wazuh/pull/672)

Hope you find it useful. Thanks for your feedback. Let me close this issue
Best regards.

**Comment by branchnetconsulting** [comment]
Great, I look forward to trying it out in 3.6.  Thanks!
